camel/typing.py

```diff
@@ -73,14 +78,18 @@ def token_limit(self) -> int:
         Returns:
             int: The maximum token limit for the given model.
         """
-        if self is ModelType.GPT_3_5_TURBO:
-            return 4096
-        elif self is ModelType.GPT_3_5_TURBO_16K:
-            return 16384
-        elif self is ModelType.GPT_4:
+        if self is ModelType.GPT_4:
             return 8192
         elif self is ModelType.GPT_4_32k:
             return 32768
+        elif self is ModelType.GPT_4_TURBO:
+            return 128000
+        elif self is ModelType.GPT_4_TURBO_VISION:
+            return 128000
+        elif self is ModelType.GPT_3_5_TURBO:
+            return 16385
+        elif self is ModelType.GPT_3_5_TURBO_16k:
+            return 16385
```

Comment:
Prefer moving the `GPT_3_5_TURBO` and `GPT_3_5_TURBO_16k` to the top.

Reply:
> Prefer moving the `GPT_3_5_TURBO` and `GPT_3_5_TURBO_16k` to the top.

done